clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -0,0 +1,56 @@
+package io.treeverse.gc
+
+import io.treeverse.clients.APIConfigurations
+import io.treeverse.clients.ApiClient
+import io.treeverse.clients.LakeFSContext._
+import io.treeverse.clients.StorageClientType
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+
+object UncommittedGarbageCollector {
+  final val UNCOMMITTED_GC_SOURCE_NAME = "uncommitted_gc"
+
+  lazy val spark = SparkSession.builder().appName("UncommittedGarbageCollector").getOrCreate()
+
+  def getDataLocation(storageNamespace: String): String = {
+    return s"${storageNamespace}/"
+  }
+
+  def main(args: Array[String]) {
+    val repo = args(0)
+    val hc = spark.sparkContext.hadoopConfiguration
+    val apiURL = hc.get(LAKEFS_CONF_API_URL_KEY)
+    val accessKey = hc.get(LAKEFS_CONF_API_ACCESS_KEY_KEY)
+    val secretKey = hc.get(LAKEFS_CONF_API_SECRET_KEY_KEY)
+    val connectionTimeout = hc.get(LAKEFS_CONF_API_CONNECTION_TIMEOUT_SEC_KEY)
+    val readTimeout = hc.get(LAKEFS_CONF_API_READ_TIMEOUT_SEC_KEY)
+    val apiConf =
+      APIConfigurations(apiURL,
+                        accessKey,
+                        secretKey,
+                        connectionTimeout,
+                        readTimeout,
+                        UncommittedGarbageCollector.UNCOMMITTED_GC_SOURCE_NAME
+                       )
+    val apiClient = ApiClient.get(apiConf)
+
+    var storageNamespace = apiClient.getStorageNamespace(repo, StorageClientType.HadoopFS)
+    if (!storageNamespace.endsWith("/")) {
+      storageNamespace += "/"
+    }
+    val dataLocation = getDataLocation(storageNamespace)
+    val uncommittedLocation = args(2) // TODO use actual uncommitted location
+    val outputLocation = args(1)
+
+    val dataPath = new Path(dataLocation)
+    val dataLister = new NaiveDataLister()
+
+    val dataDF = dataLister.listData(spark, dataPath)
+    val committedDF =
+      new NaiveCommittedAddressLister().listCommittedAddresses(spark, storageNamespace)
```

Comment:
Read of committed data must start after reading all uncommitted data (it should not be parallelized)